src/cloud-cli/applications/deploy-app-code.ts

```diff
@@ -92,3 +102,46 @@ export async function deployAppCode(host: string): Promise<number> {
   }
 }
 
+async function buildAppInDocker(appName: string): Promise<boolean> {
+  const logger = new GlobalLogger();
+
+  // Verify Docker is running
+  try {
+    execSync(`docker > /dev/null 2>&1`)
+  } catch (e) {
+    logger.error("Docker not found.  To deploy, please start the Docker daemon and make the `docker` command runnable without sudo.")
+    return false
+  }
+
+  const dockerFileName = `${deployDirectoryName}/Dockerfile.dbos`;
+  const containerName = `dbos-builder-${appName}`;
+
+  // Dockerfile content
+  const dockerFileContent = `
+FROM node:18-alpine3.17
+RUN apk update && apk add zip
+WORKDIR /app
+COPY . .
+RUN npm clean-install
+RUN npm run build
+RUN npm prune --omit=dev
```

Comment:
Should we also do an npm start here ? In the case of bcrypt the errors dont happen until you try to start the app

Reply:
What I have seen done for a similar project is the deploy command optionally running `pytest` on the code being deployed. We could do either a start -- but will have to figure out issues like conflicting HTTP(s) port -- or run the user's unit tests? Or come up with a very small functional test, similar to the runtime tests.

Reply:
Some errors will happen only during runtime -- like method execution. Should we separate this out of deploy into a build and test command

Reply:
I think the tests have to happen inside the container, so we can have more confidence the program can run in the cloud environment.

Reply:
We probably shouldn't try to run tests in deploy--too much configuration required.  We might want to separately provide an easy way to run tests in the build environment, though.